### DISCUSSION TITLE
How can I use display Markdown in a qmd file?

### DISCUSSION THREAD

**Discussion by sebastiansukstorf** [quarto-dev/quarto-cli#4521] [Q&A]
Hello,

I am aksing myself how I can use display Markdown in a qmd file corretly. I have this code:

    ```{python}
    #| echo: true

    # Pandas laden
    import pandas as pd 

    # csv Datei über read_csv laden
    file = "Priorisierung_der_Anforderungen.csv"
    df = pd.read_csv(file)
    Anzahl_Karten = str(len(df))
    
    from IPython.display import display, Markdown

    display(Markdown("""
    # Lesen der Daten

    Im Anforderungsworkshop wurden insgesamt {Anzahl_Karten} Anforderungskarten der Teilnehmenden in einer Excel Tabelle erfasst.
    """)).format(Anzahl_Karten)

    ```

I want to include the variable 'Anzhal_Karten' in the text. I understand that I need to use display, Markdown for this.

But I receive thids error:

    AttributeError: 'NoneType' object has no attribute 'format'

And my output looks like this:

    Lesen der Daten
    Im Anforderungsworkshop wurden insgesamt {Anzahl_Karten} Anforderungskarten der Teilnehmenden in einer Excel Tabelle erfasst.

Thank you for any tips.
Sebastian

**Comment by cscheid** [comment]
You need to be mindful of the correct object you're calling the `.format` on. You have

```python
display(Markdown("""... content""")).format(Anzahl_Karten)
```

but `format` is a method of a _string_, so needs to be called on a string:

```python
display(Markdown("""... content""".format(Anzahl_Karten)))
```

**Reply by sebastiansukstorf** [reply]
thank you
